### DISCUSSION TITLE
Randomness functions not callable from UI (wallet adapter ts library)

### DISCUSSION THREAD

**Discussion by kaouki99** [aptos-labs/aptos-developer-discussions#223] [Questions]
### Discord user ID

dawkaxx

### Describe your question in detail.

I'm using the new native on chain randomness on testnet. My function involving randomness is noted 
`#[randomness]
    entry fun play(
        player: &signer,
    ) ..`
   
I can call it succesfully from the SDK (signing from the SDK) but it doesn't work with the wallet adapter (trying to sign from Pontem/Petra). It works for functions not involving randomness so I guess it has to do with the wallet supports.

With Petra the error is : `generic error`. For Pontem, nothing pops up

Thanks for the help

LFM

### What error, if any, are you getting?

Generic error / no pop up from the wallet

### What have you tried or looked at?  Or how can we reproduce the error?

Described above

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

TypeScript SDK

### Describe your environment or tooling in detail

It works with the typescript sdk but not the typescript wallet adapter from a react app

**Comment by 0xmaayan** [comment]
Hi, to use a randomness entry function, the dapp should manually set `maxGasAmount` parameter when generating a transaction.
1) This is for security and protections from a malicious dapp.
2) The parameter value should be a number that the current connected account balance is high enough to pre-pay for it. Ideally dapp developer set this to be the max possible gas of all execution path, but you can set it to `10000` - that should work.

Here is an example of how a dapp can define the `maxGasAmount` property when they generate the transaction.
```
const transaction: InputTransactionData = {
      data: {
        function: "0x1::aptos_account::transfer",
        functionArguments: [account.address, 1]
      },
      options: { maxGasAmount: 10000 },
    };
```
https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/apps/nextjs-example/components/transactionFlow/TransactionParameters.tsx#L28-L37 

Note, the wallet itself should support and respect the given `maxGasAmount` value - I think Pontem currently supports it.

**Reply by kaouki99** [reply]
Hi, thanks for your reply.

First, very nice to see that you enforce best pratice to avoid test-and-abort and undergasing attacks.

I tried with 10_000 maxGasAmount as it was the only to make it work from signing with the sdk as well. The same payload doesn't work from the wallet adapter so I assume the issue is from the wallet support 🤔

**Reply by 0xmaayan** [reply]
what wallet did you use? and what version of the wallet-adapter package you use?

**Reply by kaouki99** [reply]
I used those and I tried with Pontem and Petra

```@aptos-labs/ts-sdk": "^1.14.0",
    "@aptos-labs/wallet-adapter-ant-design": "^2.2.6",
    "@aptos-labs/wallet-adapter-react": "^2.3.7",
    "@martianwallet/aptos-wallet-adapter": "^0.0.5",
    "@msafe/aptos-wallet-adapter": "^1.0.14",
    "petra-plugin-wallet-adapter": "^0.4.5",
    "@pontem/wallet-adapter-plugin": "^0.2.1",
```

**Reply by 0xmaayan** [reply]
Testing the wallet compatibility with the `maxGasAmount` on this testing dapp shows that Pontem should work with this property https://aptos-labs.github.io/aptos-wallet-adapter/

Do you mind sharing the code you use to build the payload and the error you are getting as a response from the wallet?

**Reply by floozK99** [reply]
Petra works now but no pop up from Pontem. Thanks anyway
